v16.6 Update

Jon-Corey opened the pull request.

cabal95 committed:
- Fixed an issue that prevented Rock.Mobile running inside iOS simulator from connecting to RealTime engine.

Commit:
- Added migration for PersonalizationSegment that adds a column for tracking segment update times

Commit:
- (CMS) Added check for descendant pages, audit details and description to PersonalizationSegment

david-j-leigh committed:
- (CMS) Fixed the Schedule Toolbox calendar export to prevent some events being ignored when importing to Google Calendar.

joshuahenninger committed:
- (Communication) Fixed issue where Obsidian Email Preference Entry block incorrectly displayed "Unsubscribe Successful" message.

joshuahenninger committed:
- (Core) Added support for payment gateways to provide the actual payment dates given a frequency, start date, and number of payments.

kylehenning committed:
- Handled Peer Review kickback

joshuahenninger committed:
- (Communication) Added a utm=email-header parameter to one-click unsubscribe email header.

david-j-leigh committed:
- (Communication) Added an option to create an Interaction when an SMS Pipeline action is run.

joshuahenninger committed:
- (Communication) Updated the mailto method of one-click unsubscribe to include the original subject line in the unsubscribe email.

PraveenMathew92 committed:
- Added SiteTypeFlags column to BlockType Table.

nairdo committed:
- Expand column size for longer Metric Detail descriptions.

cabal95 committed:
- Added initial support for filtering in-memory objects by expression rules.

cabal95 committed:
- Removed csproj changes that were accidentally included in previous commit.

PraveenMathew92 committed:
- Fixed namespace for SiteTypeFlags enum

cabal95 committed:
- Fixed null reference exception in integration tests.

cabal95 committed:
- Added unit tests for ExpressionHelper.GetAttributeMemberExpression method.

kylehenning committed:
- Updated styling for NCOA process block

kylehenning committed:
+ (CRM) Added a new NCOA Process Block that replaced the NCOA system with a more direct solution that provides greater service features at a lower cost.

jasonhendee committed:
- (Core) Increased InteractionSessionLocation Region column from 2 to 3 characters.

Kwame-Agyei committed:
- Updated FinancialPledge to record history and Saved Account List to respect giving type context.

Kwame-Agyei committed:
- Updated obsidian page route and fundraising donation list column widths to match webforms block.

bradencohen committed:
- Fix malformed connection data in Realtime engine.

jasonhendee committed:
- (Group) Improved the Obsidian Group Scheduler to only show occurrences for the specifically-selected date range, honor Group sort order and show a review message before sending confirmations.

cabal95 committed:
- Update various references to target packages instead of RockWeb/bin directly.

cabal95 committed:
- Removed unneeded NuGet package.

jasonhendee committed:
- (Core) Fixed an issue where the incorrect IP address can be used when calling the Interactions API for unauthenticated requests.

jasonhendee committed:
+ (Core) Fixed an issue where pressing "Enter" within the Obsidian grid's quick filter can cause row actions (i.e. Delete) to be accidentally triggered. (Fixes #5860)

kylehenning committed:
- Fixed an issue where the Location List Field Type would throw an error when being edited on the Form Builder

Commit:
- Removing Migration due to conflicts

Commit:
- Added a column for tracking segment update times in PersonalizationSegment

Commit:
- Updated migrations file

nairdo committed:
- Fix a small issue to allow an admin to delete a Financial Gateway component when the DLL is missing from their installation.

kylehenning committed:
- Removed old Spark Data Settings, Ncoa, and Spark Data API code which was replaced by the new Ncoa Process block and the Rock.NCOA project.

PraveenMathew92 committed:
+ (Event) Fixed issue of Group Schedule Notifications and Group Schedule Reminders not honoring the schedule exlusions. (Fixes #5880)

unlearnd committed:
- PreAlpha 20240606 Breaking Change fixes

unlearnd committed:
- PreAlpha 20240606 Migrations

unlearnd committed:
- Fixed compiler warnings and code generation.

nairdo committed:
- Fix display bug with GroupHistory's timeline view that would mash dates and years together into one string.

Kwame-Agyei committed:
- Fixed amount and date column justification on obsidian Fundraising Donation List block.

Kwame-Agyei committed:
- Updated Step Program Detail to use chart.js directly.

jasonhendee committed:
+ (Group) Fixed an issue where Group Requirements are not able to be overridden. (Fixes #5825)

joshuahenninger committed:
- (Core) Updated SignalR and block action request flows to populate the RockRequestContext.Method property.

PraveenMathew92 committed:
- Modified Connection Request Detail Webforms Block to accept Connection Request Id as IdKey

joshuahenninger committed:
- (Core) Added XML comment to RockRequestContext.HttpMethod clarifying that value is always uppercase and can be null.

cabal95 committed:
- Added ability to integration tests to use previous container when building new container.

cabal95 committed:
- Added some minor Obsidian features.

cabal95 committed:
- Code gen Obsidian view models.

cabal95 committed:
- Added ProducesResponseTypeAttribute as forward declaration for ASP.Net Core.

cabal95 committed:
- Added initial support for doing testing with a mock RockContext.

cabal95 committed:
- Added additional features to various cache objects for future performance improvements.

Kwame-Agyei committed:
- Added Measurement Classification Value property to Metric.

nairdo committed:
- (Core) Fixed a typo. (Fixes #5901)

PraveenMathew92 committed:
- Added UI for UTM settings to Page Shortlink Obsidian Block

cabal95 committed:
- Updated Obsidian blocks to support Rock as a nested application.

cabal95 committed:
- Keep VS happy.

cabal95 committed:
- Fixed the control gallery that I broke.

jasonhendee committed:
- (Communication) Improved performance of the SMS Conversations block by adding a database index and changing queries to only retrieve entity fields that are actually needed.

jasonhendee committed:
+ (Group) Fixed an issue where the Group Member Requirements Grid fails to export to Excel. (Fixes #5893)

jasonhendee committed:
- (Core) Removed the requirement for AdditionalSettingsJson category keys to end with a suffix of "Settings".

mattmarron committed:
+ (CRM) Fixed the addQuickReturn function with the defensive check. (Fixes #5906) (#5908)

shivambareria committed:
+ (CRM) Fixed Duplicate Finder Detail block to correctly exports correct data in all the fields. (Fixes #5895)

unlearnd committed:
+ (Core) Fixed a bug in the LocationList control which made it too narrow to be readable. (Fixes #5884)

david-j-leigh committed:
- (Test) Added unit tests for Google Map and KPI shortcodes.

Kwame-Agyei committed:
+ (Engagement) Fixed issue with Check-in configuration Streak Type rebuild not including nested groups. (Fixes #5785)

Kwame-Agyei committed:
+ (Core) Fixed issue with CalculateMetrics job not reporting exceptions. (Fixes #5904)

Commit:
- (CRM) Added Giving Configuration Obsidian block

unlearnd committed:
+ (Check-in) Fixed a bug where monthly recurrence of "First" day of week was not being honored. (Fixes #5894)

Commit:
- Fixed merge issue

david-j-leigh committed:
- (Test) Fixed a failing test caused by international date format.

Kwame-Agyei committed:
- Added obsidian Workflow List block.

shivambareria committed:
- (Finance) Fixed Change Account Allocation button in scheduled transaction view block.

david-j-leigh committed:
+ (Lava) Fixed template detection to correctly identify tags spanning multiple lines. (Fixes #5913)

david-j-leigh committed:
+ (Lava) Fixed the behavior of 'null' comparisons with the Lava Fluid engine.

jasonhendee committed:
- (Communication) Fixed an issue where the SMS Conversations block can time out when filtering by replies.

jasonhendee committed:
+ (Group) Fixed an issue where the Obsidian Group Schedule Toolbox block would show inactive Schedules when adding Group Preference Assignments. (Fixes #5294)

Kwame-Agyei committed:
- Updates to the peer network model.

Kwame-Agyei committed:
- Fixed issue with obsidian webfarm settings block not showing accurate status.

david-j-leigh committed:
+ (Group) Fixed the Group Attendance List block to correctly display custom columns. (Fixes #5917)

unlearnd committed:
- Add LMS models.

unlearnd committed:
- PreAlpha 20240620 Migrations

shivambareria committed:
- Fixed Time to Live (Seconds) in Queue Detail obsidian block.

unlearnd committed:
- Fixed compiler warnings and code generation.

shivambareria committed:
- (Finance) Updated Business Detail with simplified logic for setting the address as Previous Location type if new address is provided.

cabal95 committed:
- Added some additional helpers to unit tests.

cabal95 committed:
- Added ability to get null from additional settings that don't exist yet.

cabal95 committed:
- Fixed incorrect some XML comments.

jasonhendee committed:
+ (Finance) Fixed an issue where a Financial Transaction Detail's processing fee cannot be manually added or edited. (Fixes #5889)

david-j-leigh committed:
- Modified the Obsidian date format function to respect browser culture settings.

shivambareria committed:
- FIxed Rest Key Detail obsidian block to display error message as type of warning on save.

shivambareria committed:
- Updated Site List Obsidian Block with minor tweek.

shivambareria committed:
- Fixed media player control to check for valid index before using splice method in ts.

PraveenMathew92 committed:
- Converted Financial Gateway Detail Obsidian block to the new format.

Kwame-Agyei committed:
- Added obsidian Auth Client List block.

Kwame-Agyei committed:
- Added obsidian Auth Scope List block.

cabal95 committed:
- Fixed Obsidian ValueList and KeyValueList not performing required checks correctly.

kylehenning committed:
- Text format and spelling changes to NCOA Process Block.

shivambareria committed:
- Updated Campus Obsidian Field Type to show any inactive item if Include Inactive is marked as false and value is preselected.

Kwame-Agyei committed:
+ (Connection) Updated Bulk Update Request to allow selecting multiple campuses as a filter.

Kwame-Agyei committed:
- Updated Calendar Dimension Settings to ensure start date is at least 120 years ago.

Kwame-Agyei committed:
- Updated obsidian Workflow CreatedDateTime column.

Kwame-Agyei committed:
- Converted WebFarmNodeDetail block to new pattern.

Kwame-Agyei committed:
- Commented out Webfarm Node detail edit button.

bradencohen committed:
- Update Reminder Edit block to allow you to assign a reminder to a specific person.

Kwame-Agyei committed:
- Update Auth Client and Auth Scope Preference Key names.

Kwame-Agyei committed:
- (Group) Updated Group Detail Lava block to allow sending communication to group leaders and parents of group members.

Kwame-Agyei committed:
- Updated Group Detail Lave attribute keys.

shivambareria committed:
- (Core) Updated Campus Picker control to display the inactive campus when the inactive campus assigned as selected value on page load.

david-j-leigh committed:
+ (CMS) Updated Rock ShortLinks to make setting UTM values easier.

Commit:
- Updated Obsidian LayoutDetail block

Kwame-Agyei committed:
- Updated WebFarmNode details to read valid propeties via reflection.

Kwame-Agyei committed:
- Fixed issue with obsidian Apple Tv Page not saving if SiteId is IdKey instead of int.

joshuahenninger committed:
- (Event) Fixed issue in the Obsidian Registration Entry block where registrants were not added as group members to the event registration group linkage when the EventOccurrenceId page parameter is used.

joshuahenninger committed:
- Updated query in Obsidian Registration Entry block that retrieves the event registration linkage group.

joshuahenninger committed:
- (Event) Fixed issue in the Obsidian Registration Entry block where the Family Term setting was not being used.

Kwame-Agyei committed:
- Fixed issue with Registration Template Detail not showing form field names.

kylehenning committed:
- Handled kickback and polished NCOA process code.

Commit:
- (CMS) Added Obsidian Layout List Block

Commit:
- Updated Obsidian PersonColumn, added option to disable sorting in any column in the Grid

kylehenning committed:
- Process elevated security job now updates AccountProtectionProfile based on family members.

Commit:
- Updated PersistedDatasetDetail block to only show Named Schedules under new system category Persisted Dataset Schedules

Also took care of a small issue in Obsidian layout list block

jasonhendee committed:
+ (Communication) Improved performance of the steps taken to prepare a Communication Recipient list in order to prevent SQL timeouts when sending the Communication. (Fixes #5651)

kylehenning committed:
- Polished the Person Signal List obsidian block.

joshuahenninger committed:
- (Event) Fixed issue in Obsidian Registration Entry block where payment plan frequencies are not reset after entering an invalid then valid start date.

joshuahenninger committed:
- (Event) Fixed issue in the Obsidian Registration Entry block where payment frequencies were not updating dynamically.

joshuahenninger committed:
- (Event) Enhanced the Obsidian Registration Entry block with Start Date field validation.

Kwame-Agyei committed:
- Updated obsidian Note Type Detail to validate max result depth.

Commit:
- Updated Pledge List block formatting and fixed a date column issue

Commit:
- Moved Volunteer Generosity block form /Reporting to /Finance

cabal95 committed:
- Added some new check-in logic.

Commit:
- (Engagement) Added Campaign List Block

- Updated Campaign List Block, fixing some feedback issues

Commit:
- (Engagement) Added Obsidian Campaign List Block

Commit:
- (CMS) Added Time To Update (ms) column to Personalization Segment list block

Commit:
- Updated formatting of time duration

shivambareria committed:
- Fixed Electronic signature control to check for valid typed signature. It should not allow multiple spaces to count as an authentic signature.

bradencohen committed:
- Fix Apple TV apps not properly initializing standard Javascript.

bradencohen committed:
- Fix Copy to Clipboard button in Apple TV Page Detail block showing for new pages.

cabal95 committed:
- Added migration for check-in label model and fixed REST controller registration with duplicate names.

cabal95 committed:
- Updated some RestActionGuid values to match develop branch.

david-j-leigh committed:
- (Workflow) Modified some Workflow email actions to align with the default behavior for resolving relative links using the public website.

david-j-leigh committed:
- (Test) Added some tests for the PageRoute Lava filter.

cabal95 committed:
- Fixed unit test for Obsidian Toggle button.

jasonhendee committed:
- (Core) Fixed malformed XML comments that were causing a build warning.

PraveenMathew92 committed:
- Removed Legacy Rock.MyWell gateway from Rock Solution

kylehenning committed:
- Updated process elevated security job to update children as well.

joshuahenninger committed:
- (Event) Fixed issue in the Registration Entry block where an invalid error message is shown when a payment plan amount greater than the balance is entered.

Kwame-Agyei committed:
- (Reporting) Enhanced Service Metric Entry campus and week selection.

Kwame-Agyei committed:
- Converted Rest Key Detail block to new pattern.

Kwame-Agyei committed:
- Added obsidian User Login List block.

Kwame-Agyei committed:
- Added missed changes.

nairdo committed:
- Small style polishing on the ServiceMetricEntry block's Campus label when only one item is configured.

PraveenMathew92 committed:
- Added Shortlink Obsidian Block

jasonhendee committed:
- (Reporting) Removed the "Result" item term from the Obsidian Dynamic Data block's grid.

Commit:
- Updated GivingConfiguration Obsidian block

Commit:
- Added QueueList Obsidian block to v1.16.6

joshuahenninger committed:
- (Event) Fixed issues in the Obsidian Registration Entry block where lock existing values was not working for group member attributes, and the registrant family member could be changed for existing registrations.

joshuahenninger committed:
- (Reporting) Fixed issue in Service Metrics Entry block where person preferences were not being saved.

david-j-leigh committed:
- (Core) Fixed the DatePicker controls to correctly handle date formats with abbreviated month names.

Kwame-Agyei committed:
- Fixed issue with Registration Entry block not using Registration name as page title if event has not yet started.

Kwame-Agyei committed:
- Updated obsidian RouteDetail block to navigate to parent page after save.

joshuahenninger committed:
- (Event) Fixed an issue in the Obsidian Registration Entry block where the registrant step would not progress forward if a signature document was required.

Commit:
- Updating RIS items on Obsidian PersistedDatasetList block

joshuahenninger committed:
- Added credit card icon and 16 max length to the Obsidian Test Gateway control.

david-j-leigh committed:
+ (Core) Fixed an issue where some components are initialized with incorrect culture settings. (Fixes #5899)

cabal95 committed:
- Enhanced printing support.

kylehenning committed:
- Added the Edit Payment Plan security verb to the Registration Detail Block.

cabal95 committed:
- Updated Device model for enhanced printing support.

kylehenning committed:
- Fixed filtering process for the Group Leader Pending Notifications Job.

jasonhendee committed:
- (Core) Added EntityIntent model.

PraveenMathew92 committed:
- Fixed issues in Service Job Detail Obsidian Block.

Kwame-Agyei committed:
- Fixed issue with Giving Context Type Picker not showing fully in dropdown mode.

Kwame-Agyei committed:
- Fixed issue with Defined Value List not showing defined values if hashed id is used.

Kwame-Agyei committed:
- Fixed issue with Tag security not showing after save.

david-j-leigh committed:
- (Test) Added some tests for Communications.

david-j-leigh committed:
- (Test) Removed unused SampleData class.

shivambareria committed:
+ (Core) Fixed the toggle control in Obsidian to make the label order consistent with web forms. (Fixes #5927)

Kwame-Agyei committed:
- Added obsidian Event Occurrence List block.

shivambareria committed:
+ (Core) Fixed sorting in Signature Document List. (Fixes #5928)

PraveenMathew92 committed:
- Added Chop Job for 16.6 blocks

shivambareria committed:
- Fixed Financial batch detail block save and edit.

unlearnd committed:
- PreAlpha 20240705 Migrations

Commit:
- Added new security setting to disable predictable ID's, updated code base to work with new setting

- Added updates for the FileUrlHelper class and subsequently each file that used the class

Commit:
- Added new security setting to disable predictable ID's, updated code base to work with new setting

unlearnd committed:
- Fixed compiler warnings and code generation.

Commit:
- Added fix for filter files

unlearnd committed:
- Remove erroneously included CodeGenerated RestControllers for LMS.

joezimjs committed:
- Fix input group addons displaying wrong in form filters.

joshuahenninger committed:
- (Event) Updated the Obsidian Registration Entry block to remove costs for registrants who are wait-listed when max capacity is reached during registration.

david-j-leigh committed:
- Fixed some exceptions when running the RockCleanup job in a test environment.

kylehenning committed:
- Added block settings and filters for CalendarLava and EventItemOccurrenceListByAudienceLava.

kylehenning committed:
- Fixed typo in EventItemOccurrenceListByAudienceLava

Commit:
- Updates to various blocks after verifying functionality

joshuahenninger committed:
- (Event) Updated "Save account information for future payments" label in Obsidian Registration Entry block.

joshuahenninger committed:
- (Event) Updated the "Set Up Payment Plan" button size and label in the Obsidian Registration Entry block.

cabal95 committed:
- Fixed REST controller registration.

joshuahenninger committed:
- (Event) Updated the Obsidian Registration Entry block to use "Registrant", "Registration", "Fee", and "Discount Code" terms defined in the Registration Template.

Kwame-Agyei committed:
+ (CRM) Fixed issue with Family Pre Registration Block not allowing form submission if requried attributes are used and second person details are not provided. (Fixes #5922)

Kwame-Agyei committed:
- Updated GivingOverview block to show a 0% increase in 90 day growth if user has made no contributions in the past 90 days.

david-j-leigh committed:
+ (Jobs) Fixed the Send Note Notifications Job to notify for replies having a different Note Type. (Fixes #5790)

Commit:
- Added filter to get avatar Urls

david-j-leigh committed:
- (Test) Added tests for the Send Note Notifications Job.

PraveenMathew92 committed:
- Code Generation, Migration rollups, and version update for 16.6 alpha packaging.

cabal95 committed:
- Fixed error when only one check-in schedule available.

unlearnd committed:
- Improve support of CREATE INDEX statement for CommunicationRecipient index.

(cherry picked from commit 16af2fb979ea5e95501b94516cc58ad389e4c057)

PraveenMathew92 committed:
- Migration Fix

bradencohen committed:
- Fix Public Application Root not being included in mobile navigation bar image.

jasonhendee committed:
- (Finance) Fixed an issue where the the Scheduled Transaction List Liquid block doesn't display Financial Scheduled Transactions whose Transaction Type is null.

PraveenMathew92 committed:
- Version updated to 1.16.6.2

bradencohen committed:
- Fix missing public application root from mobile menu item images.

shivambareria committed:
- Updated Service Job Detail to make job selector dropdown Enhance for Long List

PraveenMathew92 committed:
- Added the logic to add Financial Batch Detail to the Context Entity on Save

PraveenMathew92 committed:
- Excluded Service Job Detail block from chop list

unlearnd committed:
- Improve ServiceMetricsEntry block to include time zone data for weekend dates.

PraveenMathew92 committed:
- Version updated to 1.16.6.3

cabal95 committed:
- Fix Universal tree picker WebForms implementation with nested selected items.

(cherry picked from commit a8989d9cdc6aea97fedf2acf161c3c9d22dbf479)

jasonhendee committed:
+ (Lava) Fixed an issue where the Chart Lava Shortcode did not work when used within the Obsidian Dynamic Data block. (Fixes #5944)

PraveenMathew92 committed:
- Toggle off Service Job Detail Block

unlearnd committed:
- Freeze migrations in 16.6.

jasonhendee committed:
- (Lava) Fixed an issue where attempting to retrieve the Rock request context from a Lava render context can throw an null reference exception.

shivambareria committed:
- (Event) Additional change to reduce the scope of this problem. (#5851)

PraveenMathew92 committed:
- Version updated to 1.16.6.4

PraveenMathew92 committed:
- Fixed Dictionary Extension issue in 16.6

kylehenning committed:
- Restored the Process Elevated Security Job to its original functionality.

PraveenMathew92 committed:
- Version updated to 1.16.6.5

joshuahenninger committed:
- (Event) Fixed issue in Obsidian Registration Entry block where force wait-listed registrants are charged when there is no Minimum Initial Payment.

PraveenMathew92 committed:
- Version updated to 1.16.6.6

Commit:
- Fixed a bug in Transaction Detail Avatar generation.

(cherry picked from commit 920215fad3d83dce312de2d0023069e95c81abd1)

Commit:
- Fixed a bug in the Transaction Detail block's Avatar generation.

(cherry picked from commit a6258d1058b53f9000c0394c54d3497229ca83ab)

PraveenMathew92 committed:
- Temporarily Disable Payment Plans in Registration Template

PraveenMathew92 committed:
- Version updated to 1.16.6.7

PraveenMathew92 committed:
- Version updated to 1.16.6.8

Commit:
- Updated PhotoSelect to work with new security setting

(cherry picked from commit bbdfba7e63358d1d56518961cae972e4ac042cc9)

PraveenMathew92 committed:
- Version updated to 1.16.6.9

Commit:
+ (Reporting) Fixed PhotoSelect issue with not loading SVG and profile images. (Fixes ##5985)

Jon-Corey committed:
Merge remote-tracking branch 'upstream/hotfix-1.16.6' into release-1.16.6-lpc

Jon-Corey committed:
- Moved Registration Changes to New Files

Spark changes the format and names of the registration files. This commit adds our custom changes to the new files.

Jon-Corey committed:
- Fixed Errors That Caused Build Issues

Jon-Corey committed:
- Updated Conflicting Packages

Jon-Corey committed:
- Fixed an Issue That Caused Trusted Users To Not Be Able to Submit Workflow Forms

Jon-Corey committed:
- Fixed an Issue Where Tags Wouldn't Have Enough Contrast

Removed a style override now that the contrast issue has been fixed by Spark.

Jon-Corey committed:
- Fixed an Issue Where The Registration Header Would Always Be In Spanish

Jon-Corey committed:
- Reverted Some Of The Styling Changes to the Login Block

Jon-Corey committed:
- Removed User Preferences And Added Link To CSS File

Removed the obsolete user preference references. Didn't replace with the new person preferences since this block doesn't respect the preferences anyway, and in fact just overwrites them.

Added the link to the CSS file since the CSS was moved from an old file to this one.

Jon-Corey committed:
- Pulled In Registration Payment Fix From v16.7

Pulled in commit a78ce23b231cdfb210cbd835d6850a2f3b64254d from v16.7 which fixes issue #5997.

zsmith-lakepointe reviewed (dismissed):
reviewed the major changes prior to testing in beta and ensured everything was working as expected. After completing additional edits based on beta testing, I believe this is now ready for production.

Jon-Corey commented:
I found that we had added our plugins to Git, which is not a good idea for a few reasons:

- It isn't our code to maintain (unless we've made changes to it after downloading it)
- We have to dig to find the code we have changed to make sure we have our fixes/customizations after updating
- We haven't been updating GitHub when we update a plugin in Rock's UI - meaning Git and Production are out of sync
- Only half of each of the plugins is in Git since Git is configured to ignore the DLL files that most plugins have
- We can get the latest code at any point by just installing the plugin from the Rock Shop

For these reasons I will be removing all of the unused or unedited plugin code from Git. I will then update all of our modified plugin code to the latest plugin version and commit the updated code.

I am converting this PR to a draft until the process of removing these plugins is complete.

Jon-Corey committed:
- Removed Unused and Unmodified Plugin Code From Source Control

Jon-Corey committed:
- Added LPC CODE Comments To The Plugins We've Modified

Also removed the modifications from the lava tester since there is a way to accomplish the same functionality without modifying plugin code

Jon-Corey committed:
Merge branch 'main' into release-1.16.6-lpc

Jon-Corey commented:
I have removed the unmodified and unused plugins from source control, added code comments to the parts we have modified to show which code contains our modifications, and fixed a merge conflict between `main` and `release-1.16.6-lpc`.

Next step is testing and if that goes well, I will mark this PR as ready for review again.

Jon-Corey committed:
- Fixed Incorrect Placement of LPC CODE Comment Causing Compile Errors

Jon-Corey committed:
- Fixed Translations

Added two missing translations and fixed one existing translation.

Jon-Corey committed:
- Changed CSS Class to Match The New Class on The Block

The block's class was changed from "new-group-scheduler" to "group-scheduler-v2". This change changes the class in the CSS to match so that the styles are still applied.
